addon/resolvers/classic/index.js

```diff
@@ -466,7 +470,13 @@ const Resolver = EmberObject.extend({
   },
 
   _extractDefaultExport(normalizedModuleName) {
-    let module = require(normalizedModuleName, null, null, true /* force sync */);
+
+    let module;
+    if(loader.__aliases && loader.__aliases.require) {
+     module = window[loader.__aliases.require](normalizedModuleName, null, null, true /* force sync */);
+    } else {
+     module = require(normalizedModuleName, null, null, true /* force sync */);
+    }
```

Comment:
This might be a little more resilient if you extracted a variable:
```js
let module;
const req = loader.__aliases?.require ?? require
module = req(normalizedModuleName, null, null, true /* force sync */);
```